index.js

```diff
@@ -81,7 +81,7 @@ Plotly.prototype.plot = function(data, graphOptions, callback) {
                 self.streamHost = url.parse(body['stream-host']).hostname;
             }
 
-            if ( body.error.length > 0 ) {
+            if ( typeof body === 'string' || body.error.length > 0 ) {
```

Comment:
When I was getting errors earlier, the body object was a string so it had no `error` property. As a result, it was  passing through this block and the next block calls the `callback` with a lot of empty properties, which breaks stuff downline.